### ISSUE TITLE
[Next Feature - ML] Predictive RLHF Score System

### ISSUE THREAD

**Issue by thiagobutignon** [thiagobutignon/spec-kit-clean-archicteture#196]
## 🎯 Objetivo

Implementar um modelo de Machine Learning para **prever scores RLHF antes da execução**, permitindo validação proativa e feedback instantâneo sobre qualidade de código.

## 📊 Contexto Atual

O sistema RLHF atual usa regras fixas para calcular scores (-2 a +2) após execução. Queremos antecipar esses scores usando ML.

### Modelo: Gradient Boosting Regressor (XGBoost/LightGBM)
- **Input**: Features extraídas do código/template
- **Output**: Score RLHF previsto (-2.0 a +2.0)
- **Métricas**: MAE, RMSE, R²

### Features Principais
1. **Estruturais**:
   - Tipo de step (create_file, refactor_file, etc)
   - Camada (domain, data, infra, presentation, main)
   - Target (backend, frontend, fullstack)
   - Tamanho do template (linhas, tokens)
   - Complexidade ciclomática

2. **Padrões de Código**:
   - Presença de imports proibidos (axios, prisma em domain)
   - Uso de patterns (repository, factory, adapter)
   - Estrutura Clean Architecture
   - Cobertura de testes

3. **Contextuais**:
   - Histórico de scores similares
   - Taxa de sucesso do desenvolvedor
   - Horário de execução
   - Número de tentativas anteriores

## 🏗️ Arquitetura Clean Architecture

```
features/ml-prediction/
├── domain/
│   ├── entities/
│   │   ├── prediction.ts          # Score, confidence, features
│   │   └── training-sample.ts     # Dados de treinamento
│   ├── usecases/
│   │   ├── predict-rlhf-score.ts
│   │   └── train-prediction-model.ts
│   └── repositories/
│       └── ml-model-repository.ts
├── data/
│   ├── usecases/
│   │   ├── ml-predict-rlhf-score.ts
│   │   └── ml-train-model.ts
│   └── protocols/
│       └── feature-extractor.ts
├── infra/
│   ├── ml/
│   │   ├── onnx-model-adapter.ts     # ONNX Runtime
│   │   ├── feature-extractor-impl.ts
│   │   └── model-storage.ts          # Salvar/carregar modelos
│   └── cache/
│       └── prediction-cache.ts       # Cache de predições
├── presentation/
│   └── cli/
│       └── predict-score-command.ts  # regent predict <template>
└── main/
    └── factories/
        └── ml-prediction-factory.ts
```

## 📦 Tecnologias

1. **Python (Training Pipeline)**:
   - scikit-learn / XGBoost
   - pandas, numpy
   - joblib (serialização)

2. **TypeScript (Inference)**:
   - onnxruntime-node (executar modelos ONNX)
   - @tensorflow/tfjs-node (alternativa)
   - sharp (análise de AST para features)

3. **Infraestrutura**:
   - Modelos salvos em `.regent/ml/models/`
   - Features em `.rlhf/training-data.json`
   - Cache de predições

## 🔄 Pipeline de Treinamento

```python
# scripts/ml/train-rlhf-predictor.py
1. Carregar dados históricos de .rlhf/metrics.json
2. Extrair features de cada execução
3. Dividir train/test (80/20)
4. Treinar modelo XGBoost
5. Validar com cross-validation
6. Exportar para ONNX
7. Salvar em .regent/ml/models/rlhf-predictor.onnx
```

### 1. Validação Proativa
```bash
regent predict templates/backend-domain-template.regent
# Output:
# 🔮 Predicted RLHF Score: +1.8 (85% confidence)
# ⚠️  Warning: Template may violate domain layer rules
# 💡 Suggestion: Remove axios import from domain
```

### 2. CI/CD Integration
```yaml
# .github/workflows/quality-check.yml
- name: Predict Quality
  run: regent predict --all --min-score 1.0
```

### 3. Editor Integration
```json
// .vscode/tasks.json
{
  "label": "Predict Score",
  "command": "regent predict ${file}"
}
```

## 📈 Métricas de Sucesso

- **Precisão**: MAE < 0.3 pontos
- **Cobertura**: 90%+ das predições dentro de ±0.5 do score real
- **Performance**: < 100ms por predição
- **Impacto**: Reduzir 30%+ de execuções com score negativo

### Fase 1: MVP (1-2 semanas)
- [ ] Criar estrutura Clean Architecture
- [ ] Implementar extração de features básicas
- [ ] Treinar modelo simples (Linear Regression)
- [ ] Integrar predição via CLI

### Fase 2: Produção (2-3 semanas)
- [ ] Features avançadas (AST analysis)
- [ ] Modelo robusto (XGBoost + ensemble)
- [ ] Cache e otimização
- [ ] Testes e validação

### Fase 3: Evolução Contínua
- [ ] Re-treinamento automático
- [ ] A/B testing de modelos
- [ ] Feedback loop
- [ ] Monitoramento de drift

## 🔗 Dependências

- Sistema de coleta de métricas existente (`EnhancedRLHFSystem`)
- Pipeline de feature engineering a ser criado

## 📝 Notas Técnicas

- Usar ONNX para portabilidade (treinar em Python, executar em TS)
- Implementar fallback para sistema baseado em regras
- Versionamento de modelos (`.regent/ml/models/v1/, v2/, etc`)
- Logs de predição para análise de qualidade
